pkg/plugin/httproute.go

```diff
@@ -54,6 +54,9 @@ func (r *RpcPlugin) setHTTPRouteWeight(rollout *v1alpha1.Rollout, desiredWeight
 	if err != nil {
 		msg := fmt.Sprintf(GatewayAPIUpdateError, httpRoute.GetName(), err)
 		r.LogCtx.Error(msg)
+		return pluginTypes.RpcError{
+			ErrorString: err.Error(),
+		}
```

Comment:
It seems we can delete this condition as if we will return error, argo rollouts controller will print message error
It will be great if you will check it and if it works then delete, please
Can you check it, please ?

Reply:
Updated in de5559d.

I checked it and Argo Rollouts prints logs when the plugin returns an error.

```
2024-04-29T09:31:40.670Z [DEBUG] plugin.gatewayAPI: time="2024-04-29T09:31:40Z" level=error msg="error updating Gateway API \"test\": Operation cannot be fulfilled on
httproutes.gateway.networking.k8s.io \"test\": the object has been modified; please apply your changes to the latest version and try again" plugin=trafficrouter
time="2024-04-29T09:31:40Z" level=warning msg="failed to set weight via plugin: Operation cannot be fulfilled on httproutes.gateway.networking.k8s.io \"test\": the obj
ect has been modified; please apply your changes to the latest version and try again" event_reason=TrafficRoutingError namespace=default rollout=test
```